pkg/hyperstart/hyperstart.go

```diff
@@ -186,14 +188,19 @@ func (h *Hyperstart) OpenSockets() error {
 		return err
 	}
 
-	h.ctlMulticast = startCtlMonitor(h.ctl)
+	h.ctlChDone = make(chan bool)
+	h.ctlMulticast = startCtlMonitor(h.ctl, h.ctlChDone)
 
 	return nil
 }
 
 // CloseSockets closes both CTL and IO sockets.
 func (h *Hyperstart) CloseSockets() error {
 	if !h.ctlState.closed() {
+		if h.ctlChDone != nil {
+			<-h.ctlChDone
+		}
+
```

Comment:
This is a bit racy as `CloseSockets()` could be closed twice before `h.ctlState.close()` is called.
I would simply add `close (h.ctlChDone)` to `func (c *connState) close()` to fix that issue.